### ISSUE TITLE
How To: Auto-Rotate

### ISSUE THREAD

**Issue by GrantBirki** [vasturiano/react-globe.gl#127]
### How To Auto-Rotate?

Hi! I recently started using this library to make a globe of all the places my partner and I have been to. I have been trying to make the globe "auto-rotate" for a few days now without any luck. I am using React + Astro and can't seem to get it working.

Here is a permalink to the source code where I am using this - [source link](https://github.com/GrantBirki/gl/blob/2b76571a52eebbef83d03f05a20877ffed688161/src/components/widgets/GlobeView.jsx)

That permalink will display a working globe, but I am curious as to how I can make it auto-rotate in a simple way. I looked at the `clouds` example in this repo and tried to replicate that, but whenever my code reached the `.controls()` call, it failed due to it being undefined.

TL;DR: Is anyone else using Astro + React to use a spinning globe and could help me out to make the earth spin round and round? 🌍

**Comment by Pewaukee** [comment]
For anyone in the future, you can autorotate the globe using the useRef react hook.
```
const globeEl = useRef();
useEffect(() => {
    globeEl.current.controls().autoRotate = true; // set the auto-rotate
  }, []);
...
return (
    <Globe
      ref={globeEl}
... />
```
sidenote: 
If you are using typescript, as i was trying in a nextjs project, you might try this: `if (globeEl.current) globeEl.controls().autoRotate = true;` and receive the following error: `Property 'controls' does not exist on type 'MutableRefObject<undefined>'.` switch the extension of the file to `.jsx` and use the above code. This also can help to fix other errors of styling, that i tried to include here: 
```
element.style.color = d.color;
element.style.width = `${d.size}px`;
element.style['pointer-events'] = 'auto';
```
where properties do not exist on the object, or typescript wants types

**Comment by BangNguyen1992** [comment]
On a side note, you can define the ref as followed in Typescript

```
import Globe, { GlobeMethods } from 'react-globe.gl';

const globeEl = React.useRef<GlobeMethods>();
```

**Comment by GrantBirki** [comment]
I never managed to get it working to have my globe auto-rotate. You have to grab it with your mouse and give it a manual "flick" to rotate it. Oh well 🤷‍♂
